tests/debugging/test_debugger.py

```diff
@@ -947,3 +947,85 @@ def test_debugger_log_live_probe_generate_messages():
         assert "'foo'" == msg1["debugger.snapshot"]["evaluationErrors"][0]["message"], msg1
 
         assert not msg1["debugger.snapshot"]["captures"]
+
+
+class SpanProbeTestCase(TracerTestCase):
+    def setUp(self):
+        super(SpanProbeTestCase, self).setUp()
+        self.backup_tracer = ddtrace.tracer
+        ddtrace.tracer = self.tracer
+
+    def tearDown(self):
+        ddtrace.tracer = self.backup_tracer
+        super(SpanProbeTestCase, self).tearDown()
+
+    def test_debugger_span_probe(self):
+        from tests.submod.stuff import mutator
+
+        with debugger() as d:
+            d.add_probes(
+                create_span_function_probe(
+                    probe_id="span-probe", module="tests.submod.stuff", func_qname="mutator", tags={"tag": "value"}
+                )
+            )
+
+            mutator(arg=[])
+
+            self.assert_span_count(1)
+            (span,) = self.get_spans()
+
+            assert span.name == "dd.dynamic.span"
+            assert span.resource == "mutator"
+            tags = span.get_tags()
+            assert tags["debugger.probeid"] == "span-probe"
+            assert tags["tag"] == "value"
+
+    def test_debugger_span_not_created_when_condition_was_false(self):
+        from tests.submod.stuff import mutator
+
+        with debugger() as d:
+            d.add_probes(
+                create_span_function_probe(
+                    probe_id="span-probe",
+                    module="tests.submod.stuff",
+                    func_qname="mutator",
+                    condition=DDExpression(
+                        dsl="not(contains(arg,42))", callable=dd_compile({"not": {"contains": [{"ref": "arg"}, 42]}})
+                    ),
+                )
+            )
+
+            mutator(arg=[42])  # should not trigger span
+
+            self.assert_span_count(0)
+
+            mutator(arg=[])  # should trigger span
+
+            self.assert_span_count(1)
+            (span,) = self.get_spans()
+
+            assert span.name == "dd.dynamic.span"
+            assert span.resource == "mutator"
+            assert span.get_tags()["debugger.probeid"] == "span-probe"
+
+    def test_debugger_snap_probe_linked_to_parent_span(self):
```

Comment:
can we extend this test case to also have mutator create a span to ensure the dynamic span becomes the parent of spans created within the instrumented function? It's pretty crucial that we assert this behaviour as well.

Reply:
I have added a new test case below with a dynamic span as root. I hope this is what you asked! 🙂